crispy_forms/templatetags/crispy_forms_field.py

```diff
@@ -130,6 +130,10 @@ def render(self, context):  # noqa: C901
                     css_class += " form-control-danger"
 
             if template_pack == "bootstrap4" and not is_multivalue(field):
+                if is_select(field):
```

Comment:
We're trying to push all of the logic into the templates rather than code. (It'll make it easier to extract templates from core).

You can pass in the additional classes into the django rendered widget via the template. See the BS5 `field` template for some inspiration as I've done the errors using this technique. This avoids needing the standalone template then html layout is the same and it's just the classes that change. 

(I'm on mobile at the moment but can draft this up at some point if you need help).

Reply:
The separation between templates code and Python was my concern as well. I'll have a look at BS5 `field` template and I have an unorthodox solution that I'd like to try as well.

Reply:
I have a solution but I'm not 100% that it's what you meant (and I have a question with the preappend/postappend widgets). I'll push the code in a few hours when I'm back home :-)

Reply:
> You can pass in the additional classes into the django rendered widget via the template. See the BS5 `field` template for some inspiration as I've done the errors using this technique. This avoids needing the standalone template then html layout is the same and it's just the classes that change.

@smithdc1 is this what you were thinking?
https://github.com/django-crispy-forms/django-crispy-forms/pull/1091/commits/01405272f3b833ba403b945d62ae4b7d83a6aa9b

If not: let me know, not a problem at all!

Reply:
Yes, like this. :+1: